src/libraries/Common/src/System/Net/IPv6AddressHelper.Common.cs

```diff
@@ -187,15 +210,17 @@ internal static unsafe bool IsValidStrict(char* name, int start, ref int end)
                                 i += 2;
                                 for (; i < end; i++)
                                 {
-                                    if (!char.IsAsciiDigit(name[i]))
+                                    if (name[i] < TChar.CreateTruncating('0') || name[i] > TChar.CreateTruncating('9'))
```

Comment:
Does the JIT end up doing the "right thing" in all of these cases where IsAsciiXx has been changed to multiple explicit comparisons (rather than the subtraction/comparison approach char takes)?

Reply:
It didn't. I'd missed the way the underflow in char.IsBetween changed the results. I've swapped these references out to use either HexConverter (as per one of the next comments) or a single comparison.